baseplate/lint/example_plugin.py

```diff
@@ -0,0 +1,57 @@
+# Pylint documentation for writing a checker: http://pylint.pycqa.org/en/latest/how_tos/custom_checkers.html
+# This is an example of a Pylint AST checker and should not be registered to use
+# In an AST (abstract syntax tree) checker, the code will be represented as nodes of a tree
+# We will use the astroid library: https://astroid.readthedocs.io/en/latest/api/general.html to visit and leave nodes
+# Libraries needed for an AST checker
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+# Basic example of a Pylint AST (astract syntax tree) checker
+# Checks for variables that have been reassigned in a function. If it finds a reassigned variable, it will throw an error
+class NoReassignmentChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    # Checker name
+    name = "no-reassigned-variable"
+    # Set priority to -1
+    priority = -1
+    # Message dictionary
+    msgs = {
+        # message-id, consists of a letter and numbers
+        # Letter will be one of following letters (C=Convention, W=Warning, E=Error, F=Fatal, R=Refactoring)
+        # Numbers need to be unique and in-between 9000-9999
+        # Check https://baseplate.readthedocs.io/en/stable/linters/index.html#custom-checkers-list
+        # for numbers that are already in use
+        "W9001": (
+            # displayed-message shown to user
+            "Reassigned variable found.",
+            # message-symbol used as alias for message-id
+            "reassigned-variable",
+            # message-help shown to user when calling pylint --help-msg
+            "Ensure variables are not reassigned.",
+        )
+    }
+
+    def __init__(self, linter: PyLinter = None):
+        super().__init__(linter)
+        self.variables: set = set()
+
+    # The following two methods are how we visit and leave desired nodes from the astroid library
```

Comment:
might be good to mention that these get called for us by pylint/astroid as it walks the tree